src/ocm/contexts/FeatureSupportLevelData.tsx

```diff
@@ -0,0 +1,90 @@
+import {
+  ClusterFeatureUsage,
+  FeatureId,
+  FeatureSupportLevels,
+  SupportLevel,
+  SupportLevelMap,
+} from '../../common';
+import FeatureSupportLevelDataInterface from '../../common/contexts/FeatureSupportLevelDataInterface';
+
+class FeatureSupportLevelData implements FeatureSupportLevelDataInterface {
+  private clusterSupportLevels: SupportLevelMap = {};
+  private versionSupportLevels: SupportLevelMap = {};
+  private clusterFeatureIds: FeatureId[] = [];
+  private _isFullySupported: boolean | undefined;
+  constructor(
+    private globalFeatureSupportLevels: FeatureSupportLevels,
+    private _openshiftVersion?: string,
+    clusterFeatureUsage?: string,
+  ) {
+    if (clusterFeatureUsage) {
+      this.initClusterFeatureIds(JSON.parse(clusterFeatureUsage) as ClusterFeatureUsage);
+    }
+    this.update();
+  }
+
+  public set openshiftVersion(openshiftVersion: string | undefined) {
+    if (openshiftVersion !== this._openshiftVersion) {
+      this._openshiftVersion = openshiftVersion;
+      this.update();
+    }
+  }
+
+  public get openshiftVersion(): string | undefined {
+    return this._openshiftVersion;
+  }
+
+  public get isFullySupported() {
+    return this._isFullySupported;
+  }
+
+  public get clusterSupportLevelMap(): SupportLevelMap {
+    return this.clusterSupportLevels;
+  }
+
+  getVersionSupportLevel(featureId: FeatureId): SupportLevel | undefined {
```

Comment:
wouldn't it be better to call this function `getFeatureSupportLevel`?

Reply:
good point. This is confusing. I'm differentiating between the feature support level in the context of a specific version, this is used for the badges, and the feature support level in the context of a specific cluster, this is used for review and create and for the inline comment of the cluster networking managed with vms. I'll think of a better name